src/category_theory/opposites.lean

```diff
@@ -9,79 +9,109 @@ namespace category_theory
 
 universes v₁ v₂ u₁ u₂ -- declare the `v`'s first; see `category_theory.category` for an explanation
 
-def op (C : Type u₁) : Type u₁ := C
+def opposite (C : Type u₁) : Type u₁ := C
 
 -- Use a high right binding power (like that of postfix ⁻¹) so that, for example,
 -- `presheaf Cᵒᵖ` parses as `presheaf (Cᵒᵖ)` and not `(presheaf C)ᵒᵖ`.
-notation C `ᵒᵖ`:std.prec.max_plus := op C
+notation C `ᵒᵖ`:std.prec.max_plus := opposite C
 
-variables {C : Type u₁} [𝒞 : category.{v₁} C]
+variables {C : Type u₁}
+
+def op (X : C) : Cᵒᵖ := X
+def unop (X : Cᵒᵖ) : C := X
+@[simp] lemma unop_op (X : C) : unop (op X) = X := rfl
+@[simp] lemma op_unop (X : Cᵒᵖ) : op (unop X) = X := rfl
+
+variables (C) [𝒞 : category.{v₁} C]
 include 𝒞
 
-instance opposite : category.{v₁} Cᵒᵖ :=
-{ hom  := λ X Y : C, Y ⟶ X,
+instance opposite_category : category.{v₁} (Cᵒᵖ) :=
+{ hom  := λ X Y : Cᵒᵖ, (unop Y) ⟶ (unop X),
   comp := λ _ _ _ f g, g ≫ f,
-  id   := λ X, 𝟙 X }
+  id   := λ X, 𝟙 (unop X) }
+
+namespace category.hom
+variables {C}
+def op {X Y : C} (f : X ⟶ Y) : (op Y) ⟶ (op X) := f
+def unop {X Y : Cᵒᵖ} (f : X ⟶ Y) : (unop Y) ⟶ (unop X) := f
+
+@[simp] lemma op_id (X : C) : op (𝟙 X) = 𝟙 (category_theory.op X) := rfl
+@[simp] lemma unop_id (X : Cᵒᵖ) : unop (𝟙 X) = 𝟙 (category_theory.unop X) := rfl
+@[simp] lemma comp_op_op {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) :
+  (op g) ≫ (op f) = op (f ≫ g) := rfl
+@[simp] lemma comp_unop {X Y Z : Cᵒᵖ} (f : X ⟶ Y) (g : Y ⟶ Z) :
+  unop (f ≫ g) = (unop g) ≫ (unop f) := rfl
+
```

Comment:
One of these simp lemmas seems to be backwards, though I'm not sure which one. There are three with `op` or `unop` applied to an identity or composition on the left hand side, but presumably there should be an even number of them...